src/System.Net.Http/src/System/Net/Http/Managed/HttpProxyConnectionHandler.cs

```diff
@@ -51,6 +51,15 @@ protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessa
                 SendWithProxyAsync(proxyUri, request, cancellationToken);
         }
 
+        private async Task<HttpResponseMessage> GetConnectionAndSendAsync(HttpRequestMessage request, Uri proxyUri, CancellationToken cancellationToken)
+        {
+            Debug.Assert(proxyUri.Scheme == UriScheme.Http);
+
+            var key = new HttpConnectionKey(proxyUri.IdnHost, proxyUri.Port, null);
+            HttpConnectionPool pool = _connectionPools.GetOrAddPool(key);
+            return await pool.SendAsync(request, cancellationToken);
+        }
```

Comment:
This need not be an async method, since the only time it awaits is at the very end... it can just `return pool.SendAsync(...)`.  If this were publicly exposed, we'd want to catch any exceptions that could emerge and wrap them in a Task, but I assume everywhere this is already called from does that in one way or another, e.g. something awaiting this.